assets/js/tagsinput.js

```diff
@@ -39,6 +42,84 @@ function setupTagsInput(tagBlock) {
     importTags();
   }
 
+  const ratingsTags = ['safe', 'suggestive', 'questionable', 'explicit', 'semi-grimdark', 'grimdark', 'grotesque'];
+  const checks = {numTags: false, ratingIncluded: false};
+
+  //TODO handle this better?
+  const errorsDisplayed = {numTags: false, ratingIncluded: false};
+
+  //chosing to pull from the plain editor field, even if fancy editor
+  //  is being displayed currently. This ensures the most current tags
+  //  will be pulled, in either case. This isn't using the tags[]
+  //  object, because it won't be up-to-date if the plain editor is being used
+  function tagsValid() {
+    const tagsArr = tagsTextareaPlain.value.split(',');
+    const len = tagsArr.length;
+
+    if (len >= 3) {
+      checks.numTags = true;
+      //return false;
+    }
+    else {
+      checks.numTags = false;
+    }
+
+    //TODO is there a more efficient way to check this?
+    const intersection = tagsArr.filter(tag => {
+      return ratingsTags.indexOf(tag.trim()) > -1;
+    });
+    if (intersection.length > 0) {
+      checks.ratingIncluded = true;
+    }
+    else {
+      checks.ratingIncluded = false;
+    }
+
+    return checks.numTags === true && checks.ratingIncluded === true;
+  }
+
+  if (uploadButtonPresent) {
```

Comment:
This is leaking the scope of functionality from upload.js. However, it honestly may not necessarily belong there either, because the purpose of this code is for displaying validation errors, and likely deserves its own file.